core/src/main/java/tc/oc/pgm/variables/Variable.java

```diff
@@ -52,6 +54,14 @@ default boolean isIndexed() {
     int size();
   }
 
+  interface TrackedHolders<T extends Filterable<?>> extends Variable<T> {
+    Map<T, Double> getTrackedValues(MatchFeatureContext matchFeatureContext);
+  }
+
+  interface DefinesDefault<T extends Filterable<?>> extends Variable<T> {
+    double getDefaultValue();
+  }
```

Comment:
The last thing we need is more complexity on variables, please don't. Also you should probably not be needing MatchFeatureContext here, regular features should not be needing to interact with the context manually

Reply:
> The last thing we need is more complexity on variables, please don't.

This does not seem constructive, what is the actionable advice here? I added the "complexity" as a mechanism for the stats module to be able to source a default value for the aggregate statistic and getting the value map of the variable. Looking back I do not think I need this tracked values interface and will remove it. Let me know if you have a preferred way of sourcing a default value for the aggregate stat, i.e. I could change it so the XML element specifies a default value or if you have any other ideas I'm all ears!

Reply:
Removed these interfaces in favor of using formulas